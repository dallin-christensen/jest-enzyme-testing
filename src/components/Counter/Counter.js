import React, { Component } from 'react'

class Counter extends Component {
  state = {
    count: 0
  }
  makeIncrementer = amount => () => {
    this.setState(({ count }) => ({ count: count + amount }))
  }
  icrement = this.makeIncrementer(1)
  deccrement = this.makeIncrementer(-1)
  render() {
    const { count } = this.state
    return (
      <div>
        <p>Current Count: {count}</p>
        <button
          className='increment'
          onClick={this.icrement}>
          +++
        </button>
        <button
          className='decrement'
          onClick={this.deccrement}>
          ---
        </button>
      </div>
    )
  }
}

export default Counter
