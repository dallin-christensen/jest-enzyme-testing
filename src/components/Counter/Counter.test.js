import React from 'react'
import { shallow } from 'enzyme'
import Counter from './Counter'

describe('Counter Component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<Counter />)
    const text = wrapper.find('p').text()
    expect(text).toEqual('Current Count: 0')
  })

  it('can increment the count when the increment button is clicked', () => {
    const wrapper = shallow(<Counter />)
    const incrementBtn = wrapper.find('button.increment')
    incrementBtn.simulate('click')
    const text = wrapper.find('p').text()
    expect(text).toEqual('Current Count: 1')
  })

  it('can decrement the count when the decrement button is clicked', () => {
    const wrapper = shallow(<Counter />)
    const decBtn = wrapper.find('button.decrement')
    decBtn.simulate('click')
    const text = wrapper.find('p').text()
    expect(text).toEqual('Current Count: -1')
  })
})