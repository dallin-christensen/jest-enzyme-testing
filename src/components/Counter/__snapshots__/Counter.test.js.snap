// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component matches the snapshot 1`] = `
<div>
  <p>
    Current Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    +++
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    ---
  </button>
</div>
`;
